src/socket.c

```diff
@@ -451,25 +434,29 @@ void pthreads_socket_read(zval *object, zend_long length, zend_long flags, zval
 
 	PTHREADS_SOCKET_CHECK(threaded->store.sock);
 
+	/* overflow check */
+	if ((length + 1) < 2) {
+		RETURN_FALSE;
+	}
```

Comment:
`zend_long` is a signed type, and overflow on signed types is undefined behaviour according to the C standard. PHP has a `ZEND_LONG_MAX` macro that could be used instead.

Reply:
Maybe the term `overflow check` is not right here. I've copied this snippet from php-src and think it is used to ensure `length` is not negativ and at least `1`. Because `length <= 0`would not make sense.

Reply:
That sounds correct - perhaps the comment should be removed and the condition updated to `length < 1`.